Modules/RatMama.py

```diff
@@ -0,0 +1,178 @@
+    \s+-\s+                              #  -
+    Platform:\s*(?P<platform>\w+)        # Match platform (currently can't contain spaces)
+    \s+-\s+                              #  -
+    O2:\s*(?P<o2>.+?)                    # Match oxygen status
+    \s+-\s+                              #  -
+    Language:\s*
+    (?P<full_language>                   # Match full language text (for regenerating the line)
+    (?P<language>.+?)\s*                 # Match language name. (currently unused)
+    \(                                   # The "(" of "(en-US)"
+    (?P<language_code>.+?)               # "en"
+    (?:                                  # Optional group
+        -(?P<language_country>.+)        # "-", "US" (currently unused)
+    )?                                   # Actually make the group optional.
+    \)                                   # The ")" of "(en-US)"
+    )                                    # End of full language text
+    (?:                                  # Possibly match IRC nickname
+    \s+-\s+                              #  -
+    IRC\s+Nickname:\s*(?P<nick>[^\s]+)   # IRC nickname
+    )?                                   # ... emphasis on "Possibly"
+    )                                    # End of the main capture group
+    \s*                                  # Handle any possible trailing whitespace
+    $                                    # End of pattern
+""")
+
+
+@rule(r"^\s*Incoming Client:", case_sensitive=False, full_message=True, prefixless=True,
+      pass_match=False)
+async def handle_ratmama_announcement(ctx: Context):
+    if ctx.user.nickname.casefold() not in ('ratmama[bot]',):
+        return
+
+    message: str = ctx.words_eol[0]
+    result = re.fullmatch(ratmama_regex, message)
+    client_name: str = result.group("cmdr")
+    system_name: str = result.group("system")
+    platform_name: str = result.group("platform")
+    o2_status: bool = result.group("o2") == "OK"
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! Was {exist_rescue.system}"
+
+        if platform_name.casefold() != str(exist_rescue.platform).replace('Platforms.', '').casefold():
+            diff_response += "Platform changed! "
+
+        if o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed" if o2_status else\
+                "O2 Status changed, it is now CODE RED"
+
+        if not diff_response == "":
+            await ctx.reply(diff_response)
```

Comment:
Shouldn't `exist_rescue` get updated with the new information here?

Reply:
That's a design choice that should be debated. I personally do not like mecha changing stuff up in the background. In my opinion, mecha is a tool for dispatch and therefore the more control it leaves to dispatch, the better. However I can also see it updating it and telling you what changed.

Reply:
Have we reached a consensus on this? (@theunkn0wn1) Perhaps other doers should chime in as well.

Reply:
We never actually reached a consensus on this for some reason, the conversation started in MECHA-254 has kinda died off for no apparent reason. 

Here is my verdict on this matter:
Mecha should be allowed to upgrade in severity, but not downgrade.
In the event of a downgrade as a result of a reconnect, it should warn the dispatch and take no further action. (allow the dispatch to validate the new data and issue the appropriate commands)

1.Mecha should **not** update the system name of a returning client.
 - instead it should warn the dispatch
2. Mecha **should** allow a non-cr case to be upgraded in severity to a CR.
 - it should warn the dispatch

Reply:
I have to agree with unknown here.  Mecha should not modify a case without user interaction.  Thus a case that comes in CR should remain CR until changed by dispatch, and vice versa.  Same with any other attribute of the case.  System, Client, etc.

Reply:
As per discussion in IRC, I agree. 

Mecha should warn on any inconsistency, but not act upon it.
After all, Mecha is a dispatch **assistant**, not an autopilot.

Formalized specification in this regard:
- [ ] mecha should not modify **existing** rescues in the event of a reconnection
- [ ] mecha must be able to detect inconsistencies in a reconnection event
- [ ] mecha should emit a warning to the dispatch in the event of an inconsistency